src/nvim/option.c

```diff
@@ -2131,10 +2013,10 @@ void set_string_default(const char *name, const char_u *val)
   int opt_idx = findoption((char_u *)name);
   if (opt_idx >= 0) {
     if (options[opt_idx].flags & P_DEF_ALLOCED) {
-      free(options[opt_idx].def_val[VI_DEFAULT]);
+      free(options[opt_idx].def_val);
     }
 
-    options[opt_idx].def_val[VI_DEFAULT] = (char_u *) xstrdup((char *) val);
+    options[opt_idx].def_val = (char_u *) xstrdup((char *) val);
```

Comment:
Are we moving away from `vim_strsave()`? If not, `vim_strsave(val)` wouldn't need any casts.